### PR TITLE
Doc8 - integration of documentation style checker - Fixes #1000 

### LINKED ISSUE #1000: Adopt doc8 style checker
We have to integrate this into our CI:
https://github.com/openstack/doc8

We have a Travis job to check for missing docs files so another job in the same section is probably where this needs to go.

### PR DESCRIPTION
**Pull request by Prome88**
Initial changes for doc8 configuration and fixes to documentation.

### CODE REVIEW

**Comment by Prome88** [comment]
Known limitation, should be added to ignore list:
docs/source/changelog.rst:2: D000 Duplicate explicit target name: "release notes".
docs/source/changelog.rst:2: D000 Duplicate explicit target name: "issue #889".

Tables, probalby to be added to ignore list:
docs/source/guide/testrun.rst:195: D001 Line too long
docs/source/guide/testrun.rst:197: D001 Line too long
docs/source/guide/testrun.rst:199: D001 Line too long
docs/source/guide/testrun.rst:201: D001 Line too long
docs/source/guide/testrun.rst:203: D001 Line too long
docs/source/guide/testrun.rst:205: D001 Line too long
docs/source/guide/testrun.rst:207: D001 Line too long
docs/source/guide/testrun.rst:209: D001 Line too long
docs/source/guide/testrun.rst:211: D001 Line too long

docs/source/admin.rst:77: D001 Line too long
docs/source/admin.rst:79: D001 Line too long
docs/source/admin.rst:81: D001 Line too long

Will fix and squash into c67ad56
docs/source/installing_docker.rst:146: D002 Trailing whitespace
docs/source/installing_docker.rst:150: D002 Trailing whitespace
docs/source/installing_docker.rst:154: D002 Trailing whitespace
docs/source/installing_docker.rst:159: D002 Trailing whitespace
docs/source/installing_docker.rst:163: D002 Trailing whitespace
docs/source/installing_docker.rst:176: D002 Trailing whitespace
docs/source/installing_docker.rst:181: D002 Trailing whitespace
docs/source/installing_docker.rst:184: D002 Trailing whitespace
docs/source/installing_docker.rst:188: D002 Trailing whitespace
docs/source/installing_docker.rst:192: D002 Trailing whitespace

Have to investigate as running doc8 on my local branch on unix does not report these three errors:
docs/source/guide/report.rst:69: D002 Trailing whitespace
docs/source/guide/testrun.rst:183: D002 Trailing whitespace
docs/source/guide/testrun.rst:235: D002 Trailing whitespace

**Comment by atodorov** [comment]
> Have to investigate as running doc8 on my local branch on unix does not report these three errors:
> docs/source/guide/report.rst:69: D002 Trailing whitespace

Space on otherwise empty line

> docs/source/guide/testrun.rst:183: D002 Trailing whitespace
> docs/source/guide/testrun.rst:235: D002 Trailing whitespace

Trailing spaces at the end of the line

Try `dos2unix` but I don't see mixed newline characters in these files.


```
docs/source/admin.rst:77: D001 Line too long
docs/source/admin.rst:79: D001 Line too long
docs/source/admin.rst:81: D001 Line too long
```

This table doesn't appear quite correct IMO. Need to think how to change it.


```
Tables, probably to be added to ignore list:
docs/source/guide/testrun.rst:195: D001 Line too long
docs/source/guide/testrun.rst:197: D001 Line too long
docs/source/guide/testrun.rst:199: D001 Line too long
docs/source/guide/testrun.rst:201: D001 Line too long
docs/source/guide/testrun.rst:203: D001 Line too long
docs/source/guide/testrun.rst:205: D001 Line too long
docs/source/guide/testrun.rst:207: D001 Line too long
docs/source/guide/testrun.rst:209: D001 Line too long
docs/source/guide/testrun.rst:211: D001 Line too long
```

we're thinking to refactor these statuses as part of https://github.com/kiwitcms/Kiwi/issues/236 and this table will become unnecessary. I will see if this can be done sooner.

File: docs/source/guide/testrun.rst [hunk 1]
```diff
@@ -200,7 +206,7 @@ To execute a Test Run:
 +-------------+---------------------------------------------------------------------------------------------------------------------------+
 | |image83|   | Blocked - Test Case has a dependency that has failed.                                                                     |
 +-------------+---------------------------------------------------------------------------------------------------------------------------+
-| |image84|   | Error - Test environment has problems that prevent Test Case     execution.                                               |
+| |image84|   | Error - Test environment has problems that prevent Test Case execution.                                                   |
```
Comment (atodorov) [inline docs/source/guide/testrun.rst]: FTR this change will cause a conflict while rebasing.

File: .travis.yml [hunk 2]
```diff
@@ -35,6 +35,7 @@ jobs:
   allow_failures:
     - env: CMD=pylint
     - env: CMD=bandit_site_packages
+    - env: CMD=doc8
```
Comment (atodorov) [inline .travis.yml]: Drop this, rebase and resolve the conflicts if any.

Comment (Prome88) [reply .travis.yml]: Done.

**Pull request merged**